components/greaselion/browser/greaselion_service_impl.cc

```diff
@@ -57,19 +57,19 @@ constexpr char kRunAtDocumentStart[] = "document_start";
 // NOTE: This function does file IO and should not be called on the UI thread.
 // NOTE: The caller takes ownership of the directory at extension->path() on the
 // returned object.
-scoped_refptr<Extension> ConvertGreaselionRuleToExtensionOnTaskRunner(
-    greaselion::GreaselionRule* rule,
-    const base::FilePath& install_dir,
-    std::vector<base::ScopedTempDir>* extension_dirs) {
+std::unique_ptr<greaselion::GreaselionConvertedExtension>
+ConvertGreaselionRuleToExtensionOnTaskRunner(
+    const greaselion::GreaselionRule& rule,
+    const base::FilePath& install_dir) {
   base::FilePath install_temp_dir =
       extensions::file_util::GetInstallTempDir(install_dir);
   if (install_temp_dir.empty()) {
     LOG(ERROR) << "Could not get path to profile temp directory";
     return nullptr;
   }
 
-  base::ScopedTempDir temp_dir;
-  if (!temp_dir.CreateUniqueTempDirUnderPath(install_temp_dir)) {
+  auto temp_dir = std::make_unique<base::ScopedTempDir>();
```

Comment:
no need to use pointer, you can `std::move` that object as is

Reply:
Sounds good, will fix.